### PR TITLE
Adieu razzle

### PR DESCRIPTION
**Pull request by tutok**
### Please check if the PR fulfills these requirements ###

- [x] The commit message follows our guidelines
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

* feature - remove razzle dependency
* docs update

**What is the current behavior? (You can also link to an open issue here)**

we are using `razzle` to build end project

**What is the new behavior (if this is a feature change)?**

`falcon-client` will have built in own optimized tools to compile end project

**Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)**

YES,
* falcon-client.config.js -> bootstrap.js
* razzle.config.js -> falcon-client.build.config.js
* removed razzleFalconClientPlugin.js
* removed VERBOSE env var as it does not do anything
* added size scrpt
 

**Other information:**

### CODE REVIEW

File: examples/shop-with-blog/client/package.json [hunk 1]
```diff
@@ -4,44 +4,13 @@
...
-    ],
-    "collectCoverageFrom": [
-      "src/**/*.{js,jsx,ts,tsx}"
-    ]
-  },
   "scripts": {
     "start": "falcon-client start",
-    "start:dbg": "cross-env VERBOSE=true falcon-client start -- --inspect",
-    "start:dbg-brk": "cross-env VERBOSE=true falcon-client start -- --inspect-brk",
+    "start:dbg": "cross-env falcon-client start -- --inspect",
```
Comment (Borales) [inline examples/shop-with-blog/client/package.json]: @tutok you don't need `cross-env` for 9-10 lines

File: packages/falcon-client/package.json [hunk 2]
```diff
@@ -1,49 +1,25 @@
...
   "license": "OSL-3.0",
   "main": "index.js",
-  "repository": "https://github.com/deity-io/falcon/tree/master/packages/falcon-client",
   "bin": {
     "falcon-client": "./bin/falcon-client.js"
   },
   "scripts": {
-    "test": "cross-env RAZZLE_PUBLIC_DIR=./test/static razzle test --env=jsdom",
-    "test:coverage": "cross-env RAZZLE_PUBLIC_DIR=./test/static razzle test --env=jsdom --ci --coverage"
+    "test": "cross-env node ./bin/falcon-client test --env=jsdom",
```
Comment (Borales) [inline packages/falcon-client/package.json]: @tutok `cross-env`

File: examples/shop-with-blog/client/config/default.json [hunk 3]
```diff
@@ -1,6 +1,4 @@
 {
-  "serverSideRendering": true,
```
[the remark this answers is not in the document]
Comment (tutok) [reply examples/shop-with-blog/client/config/default.json]: @Borales I simply thought that we can skip setting [defaut values](https://github.com/deity-io/falcon/tree/adieu-razzle/packages/falcon-client#config)

[the remark this answers is not in the document]
Comment (Borales) [reply examples/shop-with-blog/client/config/default.json]: @tutok can we keep them here just as a quick reference? these are some essential settings required for the app

[the remark this answers is not in the document]
Comment (tutok) [reply examples/shop-with-blog/client/config/default.json]: Yes, I will revert it.

But, reason to have `default.json` is not to keep all possible` falcon-client` settings, but instead, to keep `shop-with-blog` base configuration for` development.json` and `production.json` transforms. 
In terms of reference - we should look into documentation I believe.

[the remark this answers is not in the document]
Comment (Borales) [reply examples/shop-with-blog/client/config/default.json]: @tutok this is an example project (should it has to show available settings without forcing developers to search for a specific setting options), of course for a real-world app - they shouldn't keep them like this.

File: packages/falcon-client/README.md [hunk 4]
```diff
@@ -115,9 +115,13 @@ Builds the app for production, and outputs to the `./build` folder.
...
 The build is minified and the filenames include the hashes. Your app is ready to be deployed!
 
+### `falcon-client size`
+
+Runs an interactive zoomable treemap of output files to visualize their size. Report will be automatically open in default browser at `http://localhost:8888`.
+
 ## API contract
 
-Application needs to have the following files `index.js`, `razzle.config.js`, and optionally `falcon-client.config.js`. Each of them should be placed in application root directory.
+Application needs to have `index.js`, and following optional configuration files `bootstrap.js` and `falcon-client.build.config.js`. Each of them should be placed in application root directory.
```
Comment (Borales) [inline packages/falcon-client/README.md]:
```suggestion
Application needs to have `index.js` file, and the following optional configuration `bootstrap.js` and `falcon-client.build.config.js`  files. Each of them should be placed into the application root directory.
```

File: packages/falcon-client/README.md [hunk 5]
```diff
@@ -227,14 +202,56 @@ Falcon Client exposes set of hooks to which you can attache custom logic:
 - `onServerInitialized(server: Koa)` - handler invoked immediately after koa server setup (when middlewares like handling errors, serving static files and routes were set up)
 - `onServerStarted(server: Koa)` - handler invoked when koa server started with no errors
 
+### `falcon-client.build.config.js`
+
+This is an optional build time configuration file which is used to setting up entire build process.
+By default support for following is turned on:
```
Comment (Borales) [inline packages/falcon-client/README.md]: Not sure if I get this sentence right.

```suggestion
By default it supports the following features:
```

File: packages/falcon-client/build-utils/paths.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+  appPackageJson: resolveApp('package.json'),
+  appBuild: resolveApp('build'),
+  appBuildPublic: resolveApp('build/public'),
+  appPublic: resolveApp('public'),
+  appManifest: resolveApp('build/assets.json'),
+
+  testsSetup: resolveApp('src/setupTests.js'),
+  appBabelRc: resolveApp('.babelrc'),
+  appEslintRc: resolveApp('.eslintrc'),
```
Comment (Borales) [inline packages/falcon-client/build-utils/paths.js]: @tutok will 57-59 lines be mentioned in the docs somehow? 
Meaning, jest config is being checked in package.json file itself, but babel and eslint - in runtime files.

Comment (tutok) [reply packages/falcon-client/build-utils/paths.js]: good point, forget about these files.

File: packages/falcon-client/build-utils/webpack/index.js [hunk 7]
```diff
@@ -78,15 +78,14 @@ module.exports.startDevServer = async () => {
...
+
   if (falconConfig.clearConsole) {
     clearConsole();
   }
-
-  Logger.log('Creating an optimized production build...');
-  if (!ifRequiredFilesExists(falconConfig)) {
-    process.exit(1);
-  }
+  Logger.log(chalk.hex('#a9cf38')('Creating an optimized production build...'));
```
Comment (Borales) [inline packages/falcon-client/build-utils/webpack/index.js]: @tutok define const color? '#a9cf38'

**Comment by mateuszwozniak** [comment]
Everything looks (and works) great - nice job! However, I miss one thing - there's no description of how to add custom webpack configuration and how to modify webpack config. I know that there is reference to razzle, but right now we don't use razzle. We have compatible plugins but that doesn't mean that users should try to use razzle plugins for everything. We should provide a section "Webpack configuration" which explains how to add custom things (any webpack plugin would be enough).

**Comment by tutok** [comment]
@mateuszwozniak I see your point. Will add an additional section into documentation with some simple example.

File: packages/falcon-client/README.md [hunk 8]
```diff
@@ -227,14 +202,160 @@ Falcon Client exposes set of hooks to which you can attache custom logic:
...
+module.exports = {
+  envToBuildIn: ['SECRET_CODE']
+};
+```
+
+### Webpack
+
+Falcon Client gives you the possibility to extend the underlying webpack configuration. You can do it via exposed plugins API (not webpack plugins). It is worth to mention that Falcon Client plugins API is [razzle](https://github.com/jaredpalmer/razzle#plugins) compatible.
+
+To use Falcon Client (or Razzle) plugin, you need to instal it in your project, and add it to `plugins` array in `falcon-client.build.config.js` file:
```
Comment (mateuszwozniak) [inline packages/falcon-client/README.md]:
```suggestion
To use Falcon Client (or Razzle) plugin, you need to install it in your project, and add it to `plugins` array in `falcon-client.build.config.js` file:
```

**Pull request merged**